packages/react/src/components/Table/tableReducer.js

```diff
@@ -554,6 +555,9 @@ export const tableReducer = (state = {}, action) => {
             isSelectAllSelected: {
               $set: view ? view.table.isSelectAllSelected : false,
             },
+            loadingMoreIds: {
```

Comment:
So, to follow up on our chat about the TABLE_REGISTER not being triggered when you change the data prop. That is because we are not doing a deep compare of the data array. But if you after appending the loaded data create a new array, e.g. just by spreading the modified one you will trigger the TABLE_REGISTER:

```
const [myData, setMyData] = useState(originalData);

onLoadMore={() => {
  setMyData((currentData) => {
    // Load and append your children to the currentData
    
    // Return a new array
    return [...currentData];
  });
}}

<StatefulTable
  data={myData}
  ...
/>
```

Reply:
I do have it in the stories, but it doesn't seems to be triggering it: 
![image](https://user-images.githubusercontent.com/22034430/133257473-6230fd6b-d602-4b62-bd30-f7cbc1039b26.png)

It seems that since I do the update of the data it just became to be the `initialData` and update directly the `state reducer` so it seems to have actually changed since it doesn't have a previous anymore.

Reply:
Ok, can you push what you have so far and I will take a look?

Reply:
You are right. We need to dispatch a new action from the StatefulTable when the data has changed. After line 125 you can add something like
```
  useEffect(() => {    
    if (view.table.loadingMoreIds?.length) {
      // Dispatch action to figure out which loading more ID that should be removed 
      dispatch(moreRowsLoaded({data: initialData}))
    }
  }, [initialData]);
```
You also need to create the new action in the reducer :-)

Reply:
UPDATE: We were able to use the TABLE_REGISTER action